api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java

```diff
@@ -241,6 +249,36 @@ void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForDwpHouseholdIde
         wiremockManager.assertThatEligibilityRequestMade();
     }
 
+    @Test
+    void shouldReturnDuplicateStatusWhenTwoEligibleClaimsForSameNino() throws JsonProcessingException, InterruptedException {
+        //Given
+        NewClaimDTO dto = aValidClaimDTO();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse
+                = anIdMatchedEligibilityConfirmedUCResponseWithAllMatchesAndDwpHouseIdentifier(DWP_HOUSEHOLD_IDENTIFIER);
+        wiremockManager.stubEligibilityResponse(identityAndEligibilityResponse);
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        //When
+        List<Future<Object>> futures = executor.invokeAll(List.of(
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class),
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class)));
+
+        List<ResponseEntity<ClaimResultDTO>> results = futures.stream().map(objectFuture -> {
+            try {
+                return (ResponseEntity<ClaimResultDTO>) objectFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toList());
+
+        //Then
+        //one transaction will succeed because it commits first.
+        assertThat(results.stream().anyMatch(r -> r.getBody().getClaimStatus() == ClaimStatus.NEW)).isTrue();
```

Comment:
This assertion will also pass if both statuses are NEW, you need to make sure that you have one with NEW and one with REJECTED.

Reply:
There is one more assert which ensures that one matches with New and other with 500

Reply:
Whilst it shouldn't be possible with our code, the two anyMatch could be asserting on the same object as they assert on separate fields. Really they should both be asserting on claimStatus and statusCode.

Reply:
You are mixing up ClaimStatus and HttpStatsCode here, you need to have two assertions for each status, you are currently doing one assertion for each status.

Reply:
how about if the first assertion is on a status code of 201 and claim status of NEW?